src/aosm/azext_aosm/common/utils.py

```diff
@@ -125,6 +132,151 @@ def check_tool_installed(tool_name: str) -> None:
         raise MissingDependency(f"You must install {tool_name} to use this command.")
 
 
+def call_subprocess_raise_output(cmd: list) -> str:
+    """
+    Call a subprocess and raise a CLIError with the output if it fails.
+
+    :param cmd: command to run, in list format
+    :raise CLIError: if the subprocess fails
+    """
+    log_cmd = cmd.copy()
+    if "--password" in log_cmd:
+        # Do not log out passwords.
+        log_cmd[log_cmd.index("--password") + 1] = "[REDACTED]"
+
+    try:
+        called_process = subprocess.run(
+            cmd, encoding="utf-8", capture_output=True, text=True, check=True
+        )
+        logger.debug(
+            "Output from %s: %s. Error: %s",
+            log_cmd,
+            called_process.stdout,
+            called_process.stderr,
+        )
+
+        return called_process.stdout
+    except subprocess.CalledProcessError as error:
+        all_output: str = (
+            f"Command: {' '.join(log_cmd)}\n"
+            f"stdout: {error.stdout}\n"
+            f"stderr: {error.stderr}\n"
+            f"Return code: {error.returncode}"
+        )
+        logger.debug("The following command failed to run:\n%s", all_output)
+        # Raise the error without the original exception, which may contain secrets.
+        raise CLIError(all_output) from None
+
+
+def clean_registry_name(registry_name: str) -> str:
+    """Remove https:// from the registry name."""
+    return registry_name.replace("https://", "")
+
+
+def push_image_from_local_registry_to_acr(
```

Comment:
I'm a bit dubious that this is actually a common util function. We only use it in the artifact class. Perhaps it would sit better there?

I think we discussed using the registry classes for both target and source registries at some point. It doesn't look like you implemented that. If you had, then this would sit naturally on a registry class, of course only being used when that instance is the target.